doc/_extensions/tool_versions.py

```diff
@@ -0,0 +1,142 @@
+"""
+Copyright (c) 2023 Nordic Semiconductor ASA
+
+SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+This extension adds a custom role, :tool-version: that can be used as a
+placeholder for the actual tool version used. The extension collects
+tool versions from the "tool-versions-{os}" and pip requirement files.
+
+The role takes one argument, which is the tool that is inquired about.
+The tool must be in all uppercase, only uses underscores and follows the
+following pattern:
+
+For tool-versions-{os} files:
+    {TOOL_NAME}_VERSION_{OS}
+where OS is one of WIN10, LINUX or DARWIN
+
+For pip requirement files:
+    {TOOL_NAME}_VERSION
+
+Examples of use:
+- :tool-version:`CMAKE_VERSION_LINUX`
+- :tool-version:`SPHINX_VERSION`
+- :tool-version:`SPHINX_NCS_THEME_VERSION`
+
+"""
+
+from docutils import nodes
+from sphinx.application import Sphinx
+from typing import List, Dict, Callable
+from sphinx.util import logging
+from pathlib import Path
+import yaml
+import re
+
+__version__ = "0.1.0"
+
+logger = logging.getLogger(__name__)
+
+NRF_SCRIPTS_BASE = Path(__file__).absolute().parents[2] / "scripts"
+ZEPHYR_SCRIPTS_BASE = Path(__file__).absolute().parents[3] / "zephyr" / "scripts"
+ZEPHYR_MCUBOOT_SCRIPTS_BASE = (
+    Path(__file__).absolute().parents[3] / "bootloader" / "mcuboot" / "scripts"
+)
+
+TOOL_VERSION_FILE_PATHS = [
+    NRF_SCRIPTS_BASE / "tools-versions-win10.txt",
+    NRF_SCRIPTS_BASE / "tools-versions-linux.txt",
+    NRF_SCRIPTS_BASE / "tools-versions-darwin.txt",
+]
+
+# Ordering matters, later entries will overwrite the earlier.
+PIP_REQUIREMENT_FILE_PATHS = [
+    ZEPHYR_SCRIPTS_BASE / "requirements-base.txt",
+    ZEPHYR_SCRIPTS_BASE / "requirements-doc.txt",
+    ZEPHYR_MCUBOOT_SCRIPTS_BASE / "requirements.txt",
+    NRF_SCRIPTS_BASE / "requirements-base.txt",
+    NRF_SCRIPTS_BASE / "requirements-doc.txt",
+    NRF_SCRIPTS_BASE / "requirements-build.txt",
+]
```

Comment:
these should come as extension config parameters

Reply:
Agreed, thanks!